### ISSUE TITLE
Vue.js: useChat tool calling support

### ISSUE THREAD

**Issue by cosbgn** [vercel/ai#2566]
### Description

I use in my `vue` frontend: 

```
import { useChat } from '@ai-sdk/vue';
const { messages, input, handleSubmit, stop, error, isLoading } = useChat()
```

In my backend:

```js
import { streamText, generateText, tool } from 'ai';
import { createOpenAI } from '@ai-sdk/openai';
import { z } from 'zod';
export default defineEventHandler(async (event) => {
	const { messages } = await readBody(event);
	const openai = createOpenAI({ apiKey:process.env.OPENAI_LEEAN })
	const result = await streamText({
		model: openai('gpt-3.5-turbo'), 
		messages,
		tools: {
			search_files: tool({
				description: 'Search files',
					parameters: z.object({
					search_term: z.string().describe('A short search query which describes the file'),
				}),
				execute: async ({ search_term }) => {
					console.log("running tools")
					return {
						search_term,
						total_files: 72 + Math.floor(Math.random() * 21) - 10,
					}
				},
			})
		}
	})
	return result.toDataStreamResponse();
})
```
It first returns:

```
{
  "id": "NqFyQ7N",
  "role": "assistant",
  "content": "",
  "createdAt": "2024-08-06T12:56:00.620Z",
  "toolInvocations": [
    {
      "state": "result",
      "toolCallId": "call_jppLulgzK3DRzZPA1ChdKqGK",
      "toolName": "search_files",
      "args": {
        "search_term": "books"
      },
      "result": {
        "search_term": "books",
        "total_files": 76
      }
    }
  ],
  "internalUpdateId": "pC4G9A8"
}
```
But then when I ask something like `so what's the number?` `useChat` cleans up the message and send back only `"content":""`

This is an example chat:

```js
{
  "messages": [
    {
      "role": "user",
      "content": "How many files do I have for \"books\"?"
    },
    {
      "role": "assistant",
      "content": ""
    },
    {
      "role": "user",
      "content": "so what's the number?"
    }
  ]
}
```

The tool response is never sent back to the AI. Ideally I would use `maxRoundTrips` and the whole thing would do it itself, but it's not supported by streaming so I'm ok with sending an empty message when I have a tool response, but it get's cleaned up.

In other words in the middle message instead of `content:''` I should have 2 messages, tool_call and tool_call_results.

**Comment by lgrammel** [comment]
Tool calling is currently not supported by Vue: https://sdk.vercel.ai/docs/ai-sdk-ui/overview#ui-framework-support

Pull requests are welcome.

**Comment by lgrammel** [comment]
https://github.com/vercel/ai/pull/3046

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Feature**  
Confidence: **High**

### Rationale
This requests a new capability for the Vue integration rather than reporting failure of documented existing support.

### Evidence
- The reporter requests tool-calling behavior in Vue's `useChat`, including preserving tool calls and results in message history.
- A maintainer states that tool calling is currently not supported by Vue and links to the framework support documentation.
- A later maintainer comment links to pull request #3046 implementing related support.
